### ISSUE TITLE
Validation: Add "learn more" per validation issue

### ISSUE THREAD

**Issue by tordans** [openstreetmap/iD#5900]
I really like the fact that iD links to external sources to learn more. Like on the tag, there is the (i) that pulls from the wiki and links to the wiki. I use that a lot to learn more about how to properly use the tag.

This pattern could …
a. help to make the validation issues a learning moment, where the linked page explains more about OSM, the tags involved and the issue at had.
b. help clarify validation issues that have more than one solution, like https://github.com/openstreetmap/iD/issues/5891 for example.

The linked pages could be markdown files inside the iD Codebase. Or links to the OSM-Wiki in a separate namespace.

**Comment by quincylvania** [comment]
I think I'm in favor of this. @bhousel is adding an info (`i`) button to issues in #6140 so we would have a place for the links to live. However, there aren't many iD-specific pages on the wiki and that may be for the best right now.

What do people think of creating pages about general QA principles/issues? Like "QA:Road Crossings" or "QA:License Compatibility". There could be subsections about how various tools detect and handle issues (like iD, JOSM, KeepRight, MapRoulette). That way there could be unified documentation on how to resolve common issues.

Some pages that list types of QA errors:

- https://wiki.openstreetmap.org/wiki/JOSM/Validator
- https://wiki.openstreetmap.org/wiki/Keep_Right
- https://wiki.openstreetmap.org/wiki/Osmose/issues
- https://wiki.openstreetmap.org/wiki/MapRoulette
- https://wiki.openstreetmap.org/wiki/OSM_Tasking_Manager/Validating_data

Cc: @1ec5 and @nyurik, who may know more about how the wiki works.

**Comment by tordans** [comment]
My thoughts where: There will be cases when the validator will not fit perfectly. Or even if it does, I will not always understand what it wants from me. In similar situations for tags, I am happy to have an easy way to get to the wiki. – And maybe this could be the first step here. Scribble:

<img width="300" alt="Bildschirmfoto 2019-04-23 um 19 53 21" src="https://user-images.githubusercontent.com/111561/56604179-b2aa0c80-6601-11e9-9114-01eef3e32fd1.png">

This has the advantage: No need to add new pages to the wiki. But the existing pages might still describe edge cases of validation, if helpful.

**Comment by nyurik** [comment]
For QA, we could introduce a new documentation section (after all, wiki is the central docs place). I would also suggest using a new type of a data item - "validation rule", which would:
* link to the description wiki pages in all available languages
* have a short label and description in every language to display in iD itself
* (optional) have some declarative properties describing the rule itself

This approach will give a permanent ID to the validation rule itself, so that many systems can re-implement it and just reference it for everyone to understand what it does.

**Comment by bhousel** [comment]
> This approach will give a permanent ID to the validation rule itself, so that many systems can re-implement it and just reference it for everyone to understand what it does.

I don't think I'm quite ready for the iD validation rules to be attached to permanent ids.  We're still changing them! 

per #6218 I'd like to merge some of the rules together into the kinds of _general_ categories that a human would be interested in ("some tags are old") rather than hyperspecific checks that a machine (or OSM pedant) would be interested in ("multipolygon tags should be on the relation and not the outer").  I don't want to burden our users with 100 checkboxes in the rule list (like osmose does for example).

**Comment by quincylvania** [comment]
> I don't think I'm quite ready for the iD validation rules to be attached to permanent ids. We're still changing them!

Agreed. This is why I suggested general-purpose QA pages where users could learn about handling issues in the abstract.
